src/test/java/io/vertx/mqtt/test/server/MqttServerKeepAliveTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.mqtt.test.server;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.mqtt.MqttClient;
+import io.vertx.mqtt.MqttClientOptions;
+import io.vertx.mqtt.MqttServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * MQTT client keep alive tests using a Vert.x MQTT server to accomodate testing.
+ */
+@RunWith(VertxUnitRunner.class)
+public class MqttServerKeepAliveTest {
+
+  private Vertx vertx;
+  private MqttServer server;
+
+  private void startServer(TestContext ctx) {
+    Async async = ctx.async();
+    server.listen(1884, "localhost", ctx.asyncAssertSuccess(server -> async.complete()));
+    async.awaitSuccess(10000);
+  }
+
+  @Before
+  public void before(TestContext ctx) {
+    vertx = Vertx.vertx();
+    server = MqttServer.create(vertx);
+  }
+
+  @After
+  public void after(TestContext ctx) {
+    vertx.close(ctx.asyncAssertSuccess());
+  }
+
+  @Test
+  public void serverWillDisconnectOnTimeout(TestContext ctx) {
+    Async async = ctx.async();
+    server.endpointHandler(endpoint -> {
+      endpoint.accept(false);
+      endpoint.pingHandler(v -> ctx.fail());
+    });
+    startServer(ctx);
+    MqttClientOptions options = new MqttClientOptions();
+    options.setAutoKeepAlive(false);    // The client will manage pings manually
+    options.setKeepAliveTimeSeconds(2); // Tell the server to disconnects the client after 3 seconds of inactivity
+    MqttClient client = MqttClient.create(vertx, options);
+    client.connect(1884, "localhost", ctx.asyncAssertSuccess(ack -> {
+      client.closeHandler(v -> {
+        async.complete();
+      });
+    }));
+    async.await(4000);
```

Comment:
this is not needed

Reply:
ok :) But I would like to keep the timeout.
The last time i was running this test and it failed, it did not stop. I havent tried to find the default timeout.
If you know a better way, feel free to change

Reply:
it's really not useful here, Vert.x Unit will anyway consider the test is not finished until async.complete() is called even though the test method returns